### DISCUSSION TITLE
How is the OIDC username generated?

### DISCUSSION THREAD

**Discussion by mpfl** [TomBursch/kitchenowl#294] [Q&A]
Luck has it that OIDC was implemented mere days before I discovered KitchenOwl. Following the docs (and having set OIDC up previously in Authelia for NextCloud and Vikunja) has led to successfully implementation for web and iOS clients. Running a public URL behind Traefik is working well as well.

My question though: How is the username generated? It does not seem to be respecting the username I have in Authelia, but instead seems to generate a username based on my real name.

**Comment by TomBursch** [comment]
You're right, I overlooked the `preferred_username` field in the [specification](https://openid.net/specs/openid-connect-core-1_0.html#StandardClaims). I will update the user creation. Currently, it just removes all spaces from the lowercase name.
